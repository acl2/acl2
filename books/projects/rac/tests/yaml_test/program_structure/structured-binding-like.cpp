#include <array>
#include <tuple>
using namespace std;

// RAC begin

tuple<int, int> foo() { return tuple<int, int>(3, 2); }

int bar() {
  int a = 0, b = foo();
  return 0;
}

// RAC end
