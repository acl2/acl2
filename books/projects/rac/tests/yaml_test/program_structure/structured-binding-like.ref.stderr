structured-binding-like.cpp:10 (13-22): Incompatible types: <int, int> cannot be cast to int
 10 |   int a = 0, b = foo();
    |              ^^^^^^^^^
note: To convert types depending on template parameters to primitive types, you should add an explicit conversion (like this ac_int<32, true>(...) to convert to int)
